src/index.ts

```diff
@@ -295,3 +382,361 @@ export const getCodecFromNullaryTag =
       ),
       name,
     )
+
+/**
+ * Supports any encoded representation of nullary member values.
+ *
+ * @example
+ * ExternallyTagged<Weather, ...> = { Sun: null } | { Rain: { mm: number } }
+ */
+type ExternallyTagged<
+  A extends Sum.AnyMember,
+  B extends MemberCodecs<A>,
+> = A extends Sum.AnyMember
+  ? // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    Record<Tag<A>, B[Tag<A>] extends t.Type<any, infer C> ? C : never>
+  : never
+
+const getExternallyTaggedMemberCodec =
+  <A extends Sum.AnyMember>(sum: Sum.Sum<A>) =>
+  <B extends Tag<A>>(k: B) =>
+  <C>(vc: t.Type<ValueByTag<A, B>, C>, name = "Sum"): t.Type<A, Record<B, C>> =>
+    new t.Type(
+      name,
+      (x): x is A =>
+        pipe(
+          unknownSerialize(x),
+          O.exists(y => y[0] === k && vc.is(y[1])),
+        ),
+      (i, ctx) =>
+        pipe(
+          // Nullary/undefined will succeed on missing key, so we need to check
+          // that manually.
+          typeof i === "object" && i !== null && k in i
+            ? E.right(i)
+            : t.failure(i, ctx, `Missing key "${k}"`),
+          E.chain(x => t.strict({ [k]: vc }).validate(x, ctx)),
+          E.map(x =>
+            Sum.deserialize(sum)([k, x[k]] as unknown as Sum.Serialized<A>),
+          ),
+        ),
+      flow(
+        Sum.serialize,
+        ([k, v]) => ({ [k]: vc.encode(v as ValueByTag<A, B>) } as Record<B, C>),
+      ),
+    )
+
+/**
+ * Derive a codec for any given sum `A` provided codecs for all its members'
+ * values, decoding and encoding to an object tagged by the sum member tag.
+ *
+ * Should the types overlap, the first valid codec will succeed.
+ *
+ * @example
+ * import * as t from "io-ts"
+ * import * as Sum from "@unsplash/sum-types"
+ * import { nullary, getExternallyTaggedCodec } from "@unsplash/sum-types-io-ts"
+ * import * as E from "fp-ts/Either"
+ *
+ * type Weather = Sum.Member<"Sun"> | Sum.Member<"Rain", number>
+ * const Weather = Sum.create<Weather>()
+ *
+ * const WeatherCodec = getExternallyTaggedCodec(Weather)({
+ *   Sun: nullary,
+ *   Rain: t.number,
+ * })
+ *
+ * assert.deepStrictEqual(
+ *   WeatherCodec.decode({ Sun: undefined }),
```

Comment:
So `WeatherCodec.decode({})` will not work right?

Reply:
Correct: https://github.com/unsplash/sum-types-io-ts/pull/22/files/20d158b5a7ccedfde9f6ef206334aef2797071d9#diff-5635fef11870e2531a490fcc34223b337358689109abfe98aae8cc94aee9d547R505